connmgr.go

```diff
@@ -134,13 +135,24 @@ func NewConnManager(low, hi int, grace time.Duration, opts ...Option) *BasicConn
 		}(),
 	}
 
+	// When we're running low on memory, immediately trigger a trim.
+	// TODO: This is racy.
+	// Use watchdog.RegisterNotifee once https://github.com/raulk/go-watchdog/pull/13 is merged.
+	watchdog.NotifyGC = cm.memoryEmergency
+
 	decay, _ := NewDecayer(cfg.decayer, cm)
 	cm.decayer = decay
 
 	go cm.background()
 	return cm
 }
 
+func (cm *BasicConnMgr) memoryEmergency() {
+	// We're low on memory.
+	// Trim connections without paying attention to the silence period.
+	cm.forceTrim()
```

Comment:
I think we need a slightly different trimmer that uses more aggressive heuristics and ignores the grace period here.

Reply:
lets also add a warning to the log when this happens.